design/encryption-spi.md

```diff
@@ -0,0 +1,166 @@
+# Encrypt / decrypt support (part II)
+
+* **Status**: Draft #1
+* **Github**: TODO
+
+## Motivation
```

Comment:
I'd like to see this proposal come with some details on best practices, and compare with DB encryption